perf-tool/src/agentOptions.cpp

```diff
@@ -0,0 +1,321 @@
+
+            error = jvmti->SetEventNotificationMode(JVMTI_DISABLE, JVMTI_EVENT_VM_OBJECT_ALLOC, (jthread)NULL);
+            check_jvmti_error(jvmti, error, "Unable to disable ObjectAlloc event.");
+        }
+        else if (!command.compare("start"))
+        { 
+            printf("Object Alloc Capability already enabled.");
+            error = jvmti->SetEventNotificationMode(JVMTI_ENABLE, JVMTI_EVENT_VM_OBJECT_ALLOC, (jthread)NULL);
+            check_jvmti_error(jvmti, error, "Unable to enable VM ObjectAlloc event notifications.");
+        }
+        else
+        {
+            invalidCommand(function, command);
+        }
+    }
+    else
+    { /* cannot generate object alloc events */
+        if (!command.compare("start"))
+        {
+            memset(&capa, 0, sizeof(jvmtiCapabilities));
+            capa.can_generate_vm_object_alloc_events = 1;
+
+            error = jvmti->AddCapabilities(&capa);
+            check_jvmti_error(jvmti, error, "Unable to init object alloc events capability");
+
+            error = jvmti->SetEventNotificationMode(JVMTI_ENABLE, JVMTI_EVENT_VM_OBJECT_ALLOC, (jthread)NULL);
+            check_jvmti_error(jvmti, error, "Unable to enable VM ObjectAlloc event notifications.");
+        }
+        else if (!command.compare("stop"))
+        {
+            printf("Obect Alloc Capability already disabled");
+            error = jvmti->SetEventNotificationMode(JVMTI_DISABLE, JVMTI_EVENT_VM_OBJECT_ALLOC, (jthread)NULL);
+            check_jvmti_error(jvmti, error, "Unable to disable ObjectAlloc event.");
+        }
+        else
+        {
+            invalidCommand(function, command);
+        }
+    }
+}
+
+void modifyMonitorStackTrace(std::string function, std::string command)
+{
+    /* enable stack trace */
+    if (!command.compare("start"))
+    {
+        setMonitorStackTrace(true);
+    }
+    else if (!command.compare("stop"))
+    {
+        setMonitorStackTrace(false);
+    }
+    else
+    {
+        invalidCommand(function, command);
+    }
+}
+
+void modifyMethodEntryEvents(std::string function, std::string command, int sampleRate)
+{
+    jvmtiError error;
+    setMethodEntrySampleRate(sampleRate);
+    if (!command.compare("stop"))
+    {
+        error = jvmti->SetEventNotificationMode(JVMTI_DISABLE, JVMTI_EVENT_METHOD_ENTRY, (jthread)NULL);
+        check_jvmti_error(jvmti, error, "Unable to disable MethodEntry event.");
+    }
+    else if (!command.compare("start"))
+    { 
+        error = jvmti->SetEventNotificationMode(JVMTI_ENABLE, JVMTI_EVENT_METHOD_ENTRY, (jthread)NULL);
+        check_jvmti_error(jvmti, error, "Unable to enable MethodEntry event notifications.");
+    }
+    else
+    {
+        invalidCommand(function, command);
+    }
+}
+
+void modifyVerboseLogSubscriber(std::string function, std::string command, int sampleRate)
+{
+    /* enable stack trace */
+    if (!command.compare("start"))
+    {
+        verboseLogSubscriber = new VerboseLogSubscriber(jvmti);
```

Comment:
If someone gives two "start" commands by mistake (or if two network clients do so), then we have a memory leak because the first one cannot be retrieved.
If the design is to have just one verboseLogSubscriber, then we must check whether `verboseSubscriber` is NULL before creating a new one. Of course, at the outer level `verboseSubscriber` must be initialized to NULL, and also set to NULL after the `delete` operation.
Probably a good idea to execute a delete when the agent is going away too.